src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -102,10 +107,13 @@ public User findById(ObjectId id) throws DataAccessException {
             if (query.size() != 0) {
```

Comment:
This is not right. If query.size() is zero you should return the list of all of the users